### ISSUE TITLE
Using `--gdnative-generate-json-api` flag causes errors

### ISSUE THREAD

**Issue by rminderhoud** [godotengine/godot#10412]
**Operating system or device - Godot version:**
4.10.0-30-generic x86_64 GNU/Linux
godot: 3a4ff34 

**Issue description:**
Launching godot with the `--gdnative-generate-json-api` flag still launches the editor. If you provide an output file name to the flag, godot returns an error in loading the scene. 

I would expect the editor to not be launched if using this flag or at the very least not use the same arg to look for a scene.

```shell
$ ./godot.x11.tools.64 --gdnative-generate-json-api foobaby.json
OpenGL ES 3.0 Renderer: GeForce GTX 780/PCIe/SSE2
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ERROR: load: No loader found for resource: res://foobaby.json
   At: core/io/resource_loader.cpp:223.
ERROR: start: Failed loading scene: res://foobaby.json
   At: main/main.cpp:1462.
WARNING: cleanup: ObjectDB Instances still exist!
```

**Comment by bojidar-bg** [comment]
Confirmed by @MednauN
> Issue seems to appear after commits 3d1c031 and 36e0a72.
> Before these two commits (at 21d53f2) everything worked fine.

**Comment by akien-mga** [comment]
Well I guess this is a good hint:
> -Disabled GDNative and GDNativeScript so build compiles again

**Comment by akien-mga** [comment]
We need better validation of incorrect flags.

**Comment by karroffel** [comment]
This will be fixed by #12586
